src/client/connection.rs

```diff
@@ -101,10 +112,20 @@ impl Connection {
             let mqtt_future = self.mqtt_future(&mut command_stream, network_request_stream, framed);
 
             match self.mqtt_io(runtime, mqtt_future) {
-                Err(true) => continue 'reconnection,
-                Err(false) => break 'reconnection,
-                Ok(_v) => continue 'reconnection,
-            }
+                Err(reconnect) => {
```

Comment:
Do we need this handling? I intended to abstract all the reconnection behavior in  `mqtt_io` method and it tells whether to connect or not.

Reply:
Did you observer any bug with that?

Reply:
Ah. Probably not. I picked the commit from `pr-notifications` and didn't notice that this is not needed.